### PR TITLE
license : add AUTHORS

### PR DESCRIPTION
**Pull request by ggerganov**
ref: #6394 

- Update copyrights to acknowledge all contributors to the project
- Generate `AUTHORS` file based on git history - will be manually updated periodically
- Same changes will be applied to [ggml](https://github.com/ggerganov/ggml) and [whisper.cpp](https://github.com/ggerganov/whisper.cpp)

### CODE REVIEW

**Review by phymbert — APPROVED** [review]
Thanks for that

File: AUTHORS [hunk 1]
```diff
@@ -0,0 +1,638 @@
...
+Stephan Walter <stephan@walter.name>
+thement <40525767+thement@users.noreply.github.com>
+Niklas Korz <niklas@niklaskorz.de>
+anzz1 <anzz1@live.com>
+Gary Linscott <glinscott@gmail.com>
+Alex Nguyen <tiendung@users.noreply.github.com>
+qunash <anzoria@gmail.com>
+Erik Scholz <Green-Sky@users.noreply.github.com>
+Qingyou Meng <meng.qingyou@gmail.com>
+slaren <2141330+slaren@users.noreply.github.com>
```
Comment (phymbert) [inline AUTHORS:39]: Who is the real @slaren  :)

**Comment by JohannesGaessler** [comment]
This would of course be more work but I think in addition to a huge list that includes everyone I think it would make sense to also add a short list for the core contributors that briefly describes what they have worked on. Even as it is I am receiving many unsolicited job offers via email and due to that some of my current llama.cpp contributions are being financially supported by a third party. I was also offered sponsorship in terms of free hardware in the past. But presumably if it were easier for third parties to understand who is contributing what to the project this would also lead to more opportunities and financial resources for project contributors, which would in turn allow them to invest more time into the project.

File: LICENSE [hunk 2]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2023 Georgi Gerganov
+Copyright (c) 2023-2024 Georgi Gerganov and the llama.cpp authors
```
Comment (JohannesGaessler) [inline LICENSE:3]:
```suggestion
Copyright (c) 2023-2024 Georgi Gerganov and the llama.cpp AUTHORs
```

To make it clear that the AUTHORs file is meant.

**Comment by JohannesGaessler** [comment]
I forgot: a short list of the core contributors would also be useful for onboarding. I personally will happily use an hour of my time explaining the project to someone who is interested in working on it but doesn't know where to start. I'm saying as much on my Github page but a more visible location in the actual repository would I think be useful.

**Comment by netrunnereve** [comment]
Are we planning to have contributors run this script and commit this file along with every PR? Otherwise this will become outdated fast. By the way for other open source projects I often just see in the MIT license something like `Copyright (c) 2023-2024 OWNER and all contributors`, which would basically acknowledge everyone who worked on the project. If someone wants to get all the names they can pull it from Git, along with the specific changes that person made and so forth.

> I forgot: a short list of the core contributors would also be useful for onboarding. I personally will happily use an hour of my time explaining the project to someone who is interested in working on it but doesn't know where to start. I'm saying as much on my Github page but a more visible location in the actual repository would I think be useful.

If you're willing to do that :smiley: it I think it would be more worthwhile to record a one hour screencast going over the project structure and recommend it to new contributors in the README. Currently a lot of information is hidden away in PR comments and the code itself.

**Comment by JohannesGaessler** [comment]
>If you're willing to do that 😃 I think it would be more worthwhile to record a one hour screencast going over the project structure and recommend it to new contributors in the README. Currently a lot of information is hidden away in PR comments and the code itself.

Text documentation >> video documentation. Documentation is also helpful but the scaling is very different. Writing documentation is going to be more efficient for onboarding thousands of devs but for onboarding < 10 it will be more efficient for me to just talk to them.

**Comment by ggerganov** [comment]
> This would of course be more work but I think in addition to a huge list that includes everyone I think it would make sense to also add a short list for the core contributors that briefly describes what they have worked on.

Sounds good - open to suggestions

> Are we planning to have contributors run this script and commit this file along with every PR?

Completely optional - we'll run the script from time to time to have a somewhat up-to-date information in the `AUTHORS` file. The `git` history is the reference anyway

**Pull request merged**